utils/kafka.go

```
@@ -14,20 +14,20 @@ func PublishMessage(kafkaURI, topic string, message string) string {
 
 	conn, err := kafka.DialLeader(context.Background(), "tcp", kafkaURI, topic, partition)
 	if err != nil {
-		log.Fatal("failed to dial leader:", err)
+		log.Println("failed to dial leader:", err)
```

I would recommend to use `log.Error()` here

@abhishekraj272 Do we want to terminate this function in case error arises here?

---

yeah, returning error would be better approach

---

@uzaxirr you mean return err? there is no function as log.Error

---

> @uzaxirr you mean return err? there is no function as log.Error


Any Error Level log is fine here, and then return an error from the function

---

@abhishekraj272 
Do we also need to do error handling [here](https://github.com/ZeStream/zestream-server/blob/dev/controllers/kafka.go#L33) if this function returns an error, Ideally this Publish function should return a `message` and an `error`, and in the controller we should have checks for this error.

What do you think?

---

this function should return the error and the response should be sent from the controller

---

@Missing-Parallax 
Do you wish to incorporate these changes in this PR itself or should i create a new issue for it?